sklearn/feature_selection/rfe.py

```diff
@@ -166,15 +178,19 @@ def _fit(self, X, y, step_score=None):
         if step_score:
             self.scores_ = []
 
+        step_one_threshold = (self.step_one_threshold if
+                              self.step_one_threshold is not None else 0)
         # Elimination
         while np.sum(support_) > n_features_to_select:
             # Remaining features
             features = np.arange(n_features)[support_]
+            n_remaining_features = np.sum(support_)
```

Comment:
Is this ever not deterministic? I.e. can we determine the sequence of feature set sizes in advance, or in a separate generator function (given X.shape[1], step, n_features_to_select, step_one_threshold, step_decay)?

Reply:
RFE is deterministic and doesn't ever change based on iteration intermediate results.  I guess it's somewhat separate from this pull request but a great idea.

Reply:
The idea is that it belongs in this PR because it isolates the added complexity of step size / feature set size calculation from the selection of features. It would mean we don't need perplexingly redundant computation like `np.sum(support_)`.

Reply:
Sorry what I meant is (and you probably saw this) is all of the code you are talking about, the loop and the redundant computation RFE has always been that way I didn't add this code in my PR.  That's why I thought it should be in a second PR but it's fine.

Reply:
Yes, but until now, the loop looked relatively straightforward :)